frame/tips/src/lib.rs

```diff
@@ -379,6 +383,31 @@ decl_module! {
 			Tips::<T>::insert(&hash, tip);
 		}
 
+		/// Remove and slash an already-open tip.
+		///
+		/// May only be called from `T::RejectOrigin`.
+		///
+		/// As a result, API will slash the finder and the deposits are lost.
+		///
+		/// Emits `TipSlashed` if successful.
+		#[weight = 10_000]
+		fn slash_tip(origin, hash: T::Hash) {
+			T::RejectOrigin::ensure_origin(origin)?;
+
+			let tip = Tips::<T>::get(hash).ok_or(Error::<T>::UnknownTip)?;
+
```

Comment:
This on the other hand is VERY important to put at the bottom.

This would be a breaking change for the transaction ordering which would actually be a breaking change for most chains.

Specifically, things like Hardware wallets (ledgers etc) depend on the order of the extrinsics in order to correctly encode and sign transactions.

When you submit a transaction to a chain, you are sending only 2 numbers to identify the extrinsic: the module, and the extrinsic number.

If you add a new dispatchable function in the middle of the other dispatchable function, then the order and number of the extrinsics will change, and this is not good.

Instead, if you place new things at the bottom of the list, then the ordering will always be backwards compatible...


So please place this at the bottom of the extrinsics list.

Reply:
Got it Shawn, moved to the end.

Reply:
Very much agreeing. An additional note for when you have to break the transaction version is to check if the next polkadot release is already breaking it or not. It would be sensible to postpone non-urgent transaction-breaking PRs to a release for which hardware developers have been notified of the change and whatnot.